### PR TITLE
miles2go userspace  update, add functions for babblepaste library, add prime_e keybard keymap

### PR DESCRIPTION
**Pull request by milestogo**
## Description
Update miles2go userspace c functions and #defines

Add a prime_e keyboard synced to this version of the userspace updates. 

Cleanups in the babblepaste library - now has callback functions for babble_modeswitch_kb/user that are called when the babble mode changes. This makes it easier to have per-keyboard LED indicators. 
Added new feature to swap 2-3 keys when the mode changes, so that the most ergonomic position for cut & paste can be preserved across OSes. 

## Types of Changes
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/primekb/prime_e/keymaps/milestogo/config.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x5052
+#define PRODUCT_ID      0x0051
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    PrimeKB
+#define PRODUCT         Prime_E
+#define DESCRIPTION     Ergo_45
```
Comment (drashna) [inline keyboards/primekb/prime_e/keymaps/milestogo/config.h:28]: You should need all of this, as it's already there, and doesn't appear to be changed. 

```suggestion
```

File: keyboards/primekb/prime_e/keymaps/milestogo/keymap.c [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    [_SYM] = LAYOUT_wrap_m2primee_trns(___________________SYM_L1__________________, ___________________SYM_R1__________________, KC_TRNS, KC_TRNS, ___________________SYM_L2__________________, ___________________SYM_R2__________________, KC_TRNS, ___________________SYM_L3__________________, KC_TRNS, ___________________SYM_R3__________________),
+
+    [_NUM] = LAYOUT_wrap_m2primee(KC_TRNS, __________40_______NUM_L1__________________, __________40_______NUM_R1__________________, KC_VOLD, KC_VOLU, KC_TRNS, __________40_______NUM_L2__________________, __________40_______NUM_R2__________________, KC_ENT, KC_TRNS, __________40_______NUM_L3__________________, KC_TRNS, __________40_______NUM_R3__________________, KC_TRNS, KC_TRNS, KC_TRNS, KC_LSFT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+    [_MOV] = LAYOUT_wrap_m2primee_trns(__________40_______MOV_L1__________________, __________40_______MOV_R1__________________, KC_TRNS, KC_CDH, __________40_______MOV_L2__________________, __________40_______MOV_R2__________________, KC_TRNS, __________40_______MOV_L3__________________, KC_TRNS, __________40_______MOV_R3__________________),
+};
```
Comment (spidey3) [inline keyboards/primekb/prime_e/keymaps/milestogo/keymap.c:49]: You may want to wrap these to match the actual layout.  It looks like cformat removed any newlines. That can be prevented by putting `// clang-format off` before the code where you want the formatting preserved, and `// clang-format on` after it.

File: keyboards/primekb/prime_e/keymaps/milestogo/keymap.c [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        writePinHigh(B1);
+      }
+
+      // in use for Colemak indicator
+      if (IS_LED_ON(usb_led, USB_LED_SCROLL_LOCK)) {
+        writePinHigh(B3);
+      } else {
+        writePinLow(B3);
+      }*/
+}
```
Comment (spidey3) [inline keyboards/primekb/prime_e/keymaps/milestogo/keymap.c:88]: Both of these do nothing; perhaps they should be removed?
```suggestion
```

**Review by spidey3 — CHANGES_REQUESTED** [review]
Sorry for the long delay looking at your PR. I've made a few suggestions, but mostly it looks good.

**Review by spidey3 — APPROVED** [review]

File: keyboards/primekb/prime_e/keymaps/milestogo/keymap.c [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+// function for layer indicator LED
+layer_state_t layer_state_set_user(layer_state_t state) {
+    // Turn on bottom LED if we are in colemak, off for qwerty.
+    if (get_highest_layer(state) == _CDH) {
+        writePinHigh(B1);
+
+    } else {
+        writePinLow(B1);
+    }
```
Comment (drashna) [inline keyboards/primekb/prime_e/keymaps/milestogo/keymap.c:168]:
```suggestion
    writePin(B1, layer_state_cmp(state, _CDH));
```

Comment (milestogo) [reply keyboards/primekb/prime_e/keymaps/milestogo/keymap.c:168]: that's nifty! Thanks.

**Review by spidey3 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**